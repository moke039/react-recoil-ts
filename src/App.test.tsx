import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import renderer from "react-test-renderer";
import App from "./App";

describe("表示確認", () => {
  it("renders title", () => {
    render(<App />);
    //screen.debug();
    //screen.getByRole("");
    const titleElement = screen.getByText(/example/i);
    expect(titleElement).toBeInTheDocument();
  });
});
describe("combobox確認", () => {
  it("renders combobox", () => {
    render(<App />);
    const filterElement = screen.getByRole("combobox");
    expect(filterElement).toBeValid();
    expect(filterElement).toHaveValue("Show All");
  });
  it("select combobox", () => {
    render(<App />);
    const filterElement = screen.getByRole("combobox");
    userEvent.selectOptions(filterElement, "Show Uncompleted");
    userEvent.selectOptions(filterElement, "Show Completed");
    expect(filterElement).toHaveValue("Show Completed");
  });
});
it("snapshot", () => {
  const component = renderer.create(<App />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
