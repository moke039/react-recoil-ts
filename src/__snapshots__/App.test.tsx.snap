// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snap shot 1`] = `
<div
  className="App"
>
  <h1>
    Recoil Example
  </h1>
  <h2>
    Learn recoill with simple todo list app
  </h2>
  <ul>
    <li>
      Total items: 
      0
    </li>
    <li>
      Items completed: 
      0
    </li>
    <li>
      Items not completed: 
      0
    </li>
    <li>
      Percent completed: 
      0
    </li>
  </ul>
  Filter:
  <select
    onChange={[Function]}
    value="Show All"
  >
    <option
      value="Show All"
    >
      All
    </option>
    <option
      value="Show Completed"
    >
      Completed
    </option>
    <option
      value="Show Uncompleted"
    >
      Uncompleted
    </option>
  </select>
  <div>
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
    <button
      onClick={[Function]}
    >
      Add
    </button>
  </div>
</div>
`;
